### DISCUSSION TITLE
Feature Request: Expose Get/Set Variable nodes to other Script Canvas graphs

### DISCUSSION THREAD

**Discussion by LB-TomaszSerwanski** [o3de/o3de#5709] [Ideas]
**Is your feature request related to a problem? Please describe.**
Exposing Script Canvas Variables to another Script is currently not possible.

**Describe the solution you'd like**
A Get Variable value / Set Variable value nodes which accept the EntityID (which contains the Script), String (the name of the Script), String (the name of the Variable), and one dynamically typed pin (the value of Variable - only on Set node). The nodes would return a Variable value obtained from desired Script.

**Describe alternatives you've considered**
Another alternative would be to create a _Get Script Canvas Component_ node which would return all the Variables exposed to that Component.

**Additional context**
At the moment, the only way to access Variables in another Script is by creating and using Script Events which is more time consuming than the solutions above.

**Comment by monroegm** [comment]
@lsemp3d @carlitosan Any thoughts about this?

**Comment by lsemp3d** [comment]
I think it's possible, there may be some light concern around coupling graphs, but I don't think it would be a big problem. 

Alternatively, if it was easier to author Script Events, would that cover the request? I've had this idea to move the Script Event authoring out of the Asset Editor and into the Script Canvas editor. Creating a Script Event would be very similar to the nodes we use to define function entry/exit points, and just like with functions, we are likely able to support their authoring inside of any SC graph.

One advantage of the Script Event is that it can target more than one graph, while getting access to a graph's variable list would need to operate solely on that single instance, which is entirely fine for many applications.

**Comment by monroegm** [comment]
@LB-TomaszSerwanski There doesn't seem to be a strong use case for this request and there is a viable workaround. Closing as won't do

**Comment by lsemp3d** [comment]
I will convert this to a discussion, I do believe there is a strong use case for improving how graphs interact and/or update their data, and I would like to hear thoughts on the question I posted above.

**Comment by LB-TomaszSerwanski** [comment]
@lsemp3d, making the creation of Script Events easier and moving it inside the Script Canvas would definitely speed up the workflow. As you stated this is also a more versatile solution, so I'd say that would cover the initial request.
